mono/metadata/threads.c

```diff
@@ -6523,6 +6523,8 @@ summarizer_state_term (SummarizerGlobalState *state, gchar **out, gchar *mem, si
 		mono_get_eh_callbacks ()->mono_summarize_managed_stack (threads [i]);
 	}
 
+	mono_create_crash_hash_breadcrumb (controlling);
```

Comment:
```suggestion
	/* The value of the breadcrumb should match the "StackHash" value written by `mono_merp_write_fingerprint_payload` */
	mono_create_crash_hash_breadcrumb (controlling);
```

The control flow in crash reporting is very hard to follow.  We should add a test that the value that is written by `mono_merp_write_fingerprint_payload` is the same as the value of this breadcrumb.